config/config/src/test/java/io/helidon/config/AbstractConfigImplTest.java

```diff
@@ -16,292 +16,105 @@
-                       "str-list-" + level + "@LIST");
-    }
-
-
-    public static List<String> nodeNames(List<Config> nodeList) {
-        return nodeList.stream()
-                .map(AbstractConfigImplTest::nodeName)
-                .collect(Collectors.toList());
-    }
-
-    public static String nodeName(Config node) {
-        return node.name() + "@" + node.type();
-    }
-
-    public static class ValueConfigBean {
-        static final ValueConfigBean EMPTY = new ValueConfigBean("EMPTY", "");
-
-        private final String meta;
-        private final String text;
-
-        public ValueConfigBean(String meta, String text) {
-            this.meta = meta;
-            this.text = text;
-        }
-
-        public String getText() {
-            return text;
-        }
-
-        static ValueConfigBean empty() {
-            return EMPTY;
-        }
-
-        public static ValueConfigBean fromString(String string) {
-            return new ValueConfigBean("fromString", string);
-        }
-
-        // unit test support to build for comparison
-        public static ValueConfigBean utFromConfig(String string) {
-            return new ValueConfigBean("fromConfig", string);
-        }
-
-        public static ValueConfigBean fromConfig(Config config) {
-            return new ValueConfigBean("fromConfig", config.asString().get());
-        }
-
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) {
-                return true;
-            }
-            if (o == null || getClass() != o.getClass()) {
+    void testConfigCorrectUnlocking() {
+        // GIVEN
+        RuntimeException exception = new RuntimeException("LOCKS UNLOCK TEST");
+
+        // a provider with a publisher that throws an exception when calling 'subscribe()'
+        ProviderImpl providerMock = mock(ProviderImpl.class);
+        Flow.Publisher<ConfigDiff> publisherMock = Mockito.mock(Flow.Publisher.class);
+        when(providerMock.changes()).thenReturn(publisherMock);
+        Mockito.doThrow(exception).when(publisherMock).subscribe(Mockito.any());
+
+        // and an almost empty stub of AbstractConfigImpl
+        AbstractConfigImpl config = configStub(
+                Config.Type.LIST,
+                mock(ConfigKeyImpl.class),
+                mock(ConfigKeyImpl.class),
+                new ConfigFactory(
+                        mock(ConfigMapperManager.class),
+                        mock(ConfigNode.ObjectNode.class),
+                        mock(ConfigFilter.class),
+                        providerMock),
+                mock(ConfigMapperManager.class));
+
+        // WHEN we reproduce issue https://github.com/oracle/helidon/issues/299
+        try {
+            config.subscribe();
+        } catch (Exception e) {
+            // THEN
+            // we get the exception we threw and not a 'java.lang.IllegalMonitorStateException'
+            assertThat(e, equalTo(exception));
+        }
+    }
```

Comment:
use `assertThrows` - this test would pass if no exception is thrown.

```
RuntimeException exception = assertThrows(RuntimeException.class, () -> config.subscribe());
assertThat(exception, sameInstance(exception));
```

Reply:
My bad, it's fixed now, thanks!